firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -900,6 +899,47 @@ private ListenerRegistration addSnapshotListenerInternal(
         firestore.getClient(), queryListener, activity, wrappedListener);
   }
 
+  @PublicApi
+  public String getPath() {
+    return query.getPath().canonicalString();
+  }
+
+  @PublicApi
+  public List<com.google.firebase.firestore.Filter> getFilters() {
+    com.google.firebase.firestore.Filter newFilter;
+    List<com.google.firebase.firestore.Filter> filters = new ArrayList<>();
+    for (Filter filter : query.getFilters()) {
+      newFilter = new com.google.firebase.firestore.Filter(filter);
+      filters.add(newFilter);
+    }
+    return filters;
+  }
+
+  @PublicApi
+  public List<OrderBy> getExplicitOrderBy() {
```

Comment:
The reason why I had named it `getExplicitOrderBy()` was to avoid conflicts with the existing `getOrderBy()`, which is only used internally. So in order to differentiate the two:
- `getOrderBy()` became `getInternalOrderBy()`
- `getExplicityOrderBy()` became `getOrderBy()`